tests/unit/query_v2_expression_evaluator.cpp

```diff
@@ -0,0 +1,1726 @@
+  ASSERT_EQ(EvaluateFunction("HEAD", argument).ValueInt(), 3);
+  argument.ValueList().clear();
+  ASSERT_TRUE(EvaluateFunction("HEAD", argument).IsNull());
+  ASSERT_THROW(EvaluateFunction("HEAD", 2), FunctionRuntimeException);
+}
+
+TEST_F(FunctionTest, Properties) {
+  ASSERT_THROW(EvaluateFunction("PROPERTIES"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("PROPERTIES", TypedValue()).IsNull());
+  const auto height = shard_manager->NameToProperty("height");
+  const auto age = shard_manager->NameToProperty("age");
+  auto v1 = CreateVertex({{height, Value(static_cast<int64_t>(5))}, {age, Value(static_cast<int64_t>(10))}},
+                         shard_manager.get());
+  auto e = CreateEdge({{height, Value(static_cast<int64_t>(3))}, {age, Value(static_cast<int64_t>(15))}},
+                      shard_manager.get());
+
+  auto prop_values_to_int = [](TypedValue t) {
+    std::unordered_map<std::string, int> properties;
+    for (auto property : t.ValueMap()) {
+      properties[std::string(property.first)] = property.second.ValueInt();
+    }
+    return properties;
+  };
+
+  ASSERT_THAT(prop_values_to_int(EvaluateFunction("PROPERTIES", v1)),
+              UnorderedElementsAre(testing::Pair("height", 5), testing::Pair("age", 10)));
+  ASSERT_THAT(prop_values_to_int(EvaluateFunction("PROPERTIES", e)),
+              UnorderedElementsAre(testing::Pair("height", 3), testing::Pair("age", 15)));
+  ASSERT_THROW(EvaluateFunction("PROPERTIES", 2), FunctionRuntimeException);
+}
+
+TEST_F(FunctionTest, Last) {
+  ASSERT_THROW(EvaluateFunction("LAST"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("LAST", TypedValue()).IsNull());
+  auto argument = MakeTypedValueList(3, 4, 5);
+  ASSERT_EQ(EvaluateFunction("LAST", argument).ValueInt(), 5);
+  argument.ValueList().clear();
+  ASSERT_TRUE(EvaluateFunction("LAST", argument).IsNull());
+  ASSERT_THROW(EvaluateFunction("LAST", 5), FunctionRuntimeException);
+}
+
+TEST_F(FunctionTest, Size) {
+  ASSERT_THROW(EvaluateFunction("SIZE"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("SIZE", TypedValue()).IsNull());
+  auto argument = MakeTypedValueList(3, 4, 5);
+  ASSERT_EQ(EvaluateFunction("SIZE", argument).ValueInt(), 3);
+  ASSERT_EQ(EvaluateFunction("SIZE", "john").ValueInt(), 4);
+  ASSERT_EQ(EvaluateFunction("SIZE",
+                             std::map<std::string, TypedValue>{
+                                 {"a", TypedValue(5)}, {"b", TypedValue(true)}, {"c", TypedValue("123")}})
+                .ValueInt(),
+            3);
+  ASSERT_THROW(EvaluateFunction("SIZE", 5), FunctionRuntimeException);
+}
+
+TEST_F(FunctionTest, StartNode) {
+  ASSERT_THROW(EvaluateFunction("STARTNODE"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("STARTNODE", TypedValue()).IsNull());
+  auto v1 = CreateVertex({}, shard_manager.get());
+  auto e = CreateEdge({}, shard_manager.get());
+  const auto expected = VertexId{{}, 0};
```

Comment:
Please also use `{}` because somebody might tricked to think `0` is some sort of id or something.